packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -5581,7 +5581,7 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
         if (entry.currentState.index <= _RouteLifecycle.idle.index) {
           // The entry may have been disposed if the pop finishes synchronously.
           assert(entry.route._popCompleter.isCompleted);
-          entry.currentState = _RouteLifecycle.popping;
+          entry.currentState = _RouteLifecycle.pop;
```

Comment:
in flush history update, pop will notify observer and then transition to popping. I can't really find out why I made this change in the first place, but I think reverting this back to pop should be fine as long as no other error